### PR TITLE
EVAKA-3899: citizen responds to decisions e2e-tests

### PR DESCRIPTION
**Pull request by Joosakur**

### CODE REVIEW

File: frontend/e2e-test/test/e2e/dev-api/types.ts [hunk 1]
```diff
@@ -471,6 +475,44 @@ export interface Decision {
...
+}
+
+export function deserializeDecision(json: JsonOf<Decision>): Decision {
+  return {
+    ...json,
+    startDate: LocalDate.parseIso(json.startDate),
+    endDate: LocalDate.parseIso(json.endDate),
+    sentDate: LocalDate.parseIso(json.sentDate)
+  }
+}
```
Comment (Joosakur) [inline frontend/e2e-test/test/e2e/dev-api/types.ts:514]: I think these are now duplicated in 3 places (citizen-frontend, employee-frontend and e2e-tests), so if there's enough other cases like this, at some point we may want to think about sharing api-typings somehow

Comment (reynders) [reply frontend/e2e-test/test/e2e/dev-api/types.ts:514]: Yes!

File: frontend/e2e-test/test/e2e/pages/citizen/citizen-decision-response.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: 2017-2020 City of Espoo
```
Comment (reynders) [inline frontend/e2e-test/test/e2e/pages/citizen/citizen-decision-response.ts:1]: 2021

File: frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts [hunk 3]
```diff
@@ -9,72 +9,248 @@ import {
...
 
-  const decision = await Fixture.decision()
-    .with({
-      applicationId: application.id,
-      employeeId: employee.data.id,
-      unitId: application.form.apply.preferredUnits[0],
-      startDate: application.form.preferredStartDate
-    })
-    .save()
+  await execSimpleApplicationActions(applicationId, [
```
Comment (reynders) [inline frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts:64]: Why not just add the decision fixtures with DecisionBuilder?

File: frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts [hunk 4]
```diff
@@ -9,72 +9,248 @@ import {
...
+    'move-to-waiting-placement',
+    'create-default-placement-plan',
+    'send-decisions-without-proposal'
+  ])
+  await runPendingAsyncJobs()
+
+  const decisions = await getDecisionsByApplication(applicationId)
+  if (decisions.length !== 2) throw Error('Expected 2 decisions')
+  const preschoolDecisionId = decisions.find((d) => d.type === 'PRESCHOOL')?.id
```
Comment (reynders) [inline frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts:73]: If decisionbuilder is used you have the id's and can use those directly

File: frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts [hunk 5]
```diff
@@ -9,72 +9,248 @@ import {
...
+  // preschool_daycare decision cannot be accepted before accepting preschool
+  await t
+    .expect(
+      citizenDecisionResponsePage
+        .submitResponseBtn(preschoolDaycareDecisionId)
+        .hasAttribute('disabled')
+    )
+    .ok()
+
+  // accept preschool decision
```
Comment (reynders) [inline frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts:128]: These commented blocks should go to the citizen-decision page object (tests in spec, page structure in page object)

File: frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts (hunk 3 above)
Comment (Joosakur) [reply frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts:64]: That won't create placement plans so accepting will fail. This approach ensures there's actually valid and realistic data. Also changes to application logic and data model won't break e2e-tests as easily.

File: frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts (hunk 4 above)
Comment (Joosakur) [reply frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts:73]: yeah that is a trade-off

File: frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts (hunk 5 above)
Comment (Joosakur) [reply frontend/e2e-test/test/e2e/specs/0_citizen/citizen-decisions.spec.ts]: added 2 commits, now better?

**Review by reynders — APPROVED** [review]

**Pull request merged**